modules/loader-utils/src/lib/request-utils/request-scheduler.ts

```diff
@@ -132,14 +136,17 @@ export default class RequestScheduler {
 
   /** We check requests asynchronously, to prevent multiple updates */
   _issueNewRequests(): void {
-    if (!this.deferredUpdate) {
-      this.deferredUpdate = setTimeout(() => this._issueNewRequestsAsync(), 0);
+    if (this.deferredUpdate !== null) {
+      clearTimeout(this.deferredUpdate);
     }
+    this.deferredUpdate = setTimeout(() => this._issueNewRequestsAsync(), this.props.debounce);
   }
 
   /** Refresh all requests  */
   _issueNewRequestsAsync() {
-    // TODO - shouldn't we clear the timeout?
+    if (this.deferredUpdate !== null) {
```

Comment:
if `deferredUpdate` a timer? should we rename it? `deferredUpdateTimer`? This reads a bit confusing as it stands.

Reply:
The slightly tricky part is that [it depends](https://stackoverflow.com/a/56239226), but I think we can use the `ReturnType<typeof setTimeout>` trick to avoid `any` here and clarify the name. Done!